src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/GetComInterfaceForObjectTests.cs

```diff
@@ -0,0 +1,233 @@
+        {
+            var o = new ClassWithInterface();
+            IntPtr iUnknown = Marshal.GetComInterfaceForObject<ClassWithInterface, NonGenericInterface>(o);
+            try
+            {
+                Assert.NotEqual(IntPtr.Zero, iUnknown);
+            }
+            finally
+            {
+                Marshal.Release(iUnknown);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void GetComInterfaceForObject_GenericWithValidStruct_ReturnsExpected()
+        {
+            var o = new StructWithInterface();
+            IntPtr iUnknown = Marshal.GetComInterfaceForObject<StructWithInterface, NonGenericInterface>(o);
+            try
+            {
+                Assert.NotEqual(IntPtr.Zero, iUnknown);
+            }
+            finally
+            {
+                Marshal.Release(iUnknown);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void GetComInterfaceForObject_NonGenericWithValidClass_ReturnsExpected()
+        {
+            var o = new ClassWithInterface();
+            IntPtr iUnknown = Marshal.GetComInterfaceForObject(o, typeof(NonGenericInterface));
+            try
+            {
+                Assert.NotEqual(IntPtr.Zero, iUnknown);
+            }
+            finally
+            {
+                Marshal.Release(iUnknown);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void GetComInterfaceForObject_NonGenericWithValidStruct_ReturnsExpected()
+        {
+            var o = new StructWithInterface();
+            IntPtr iUnknown = Marshal.GetComInterfaceForObject(o, typeof(NonGenericInterface));
+            try
+            {
+                Assert.NotEqual(IntPtr.Zero, iUnknown);
+            }
+            finally
+            {
+                Marshal.Release(iUnknown);
+            }
+        }
+
+        [Theory]
+        [InlineData(CustomQueryInterfaceMode.Allow)]
+        [InlineData(CustomQueryInterfaceMode.Ignore)]
+        [InlineData(CustomQueryInterfaceMode.Allow + 1)]
+        [InlineData(CustomQueryInterfaceMode.Ignore - 1)]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void GetComInterfaceForObject_NonGenericCustomQueryInterfaceModeWithValidClass_ReturnsExpected(CustomQueryInterfaceMode mode)
+        {
+            var o = new ClassWithInterface();
+            IntPtr iUnknown = Marshal.GetComInterfaceForObject(o, typeof(NonGenericInterface));
+            try
+            {
+                Assert.NotEqual(IntPtr.Zero, iUnknown);
+            }
+            finally
+            {
+                Marshal.Release(iUnknown);
+            }
+        }
+
+        [Theory]
+        [InlineData(CustomQueryInterfaceMode.Allow)]
+        [InlineData(CustomQueryInterfaceMode.Ignore)]
+        [InlineData(CustomQueryInterfaceMode.Allow + 1)]
```

Comment:
Please don't add the unnatural out of bounds values for the enumeration values. These values aren't defined and if and when we do define them there usage here is purely noise and adds no obvious value.

Reply:
I disagree - there is precedent for this. Across all corefx tests we always (usually) check out of bounds enums in tests to see the behaviour if the user passes in an invalid input. I know you mentioned that this is an invalid use of the API, but that is my exact point and therefore should be tested to make sure nothing weird happens!

Reply:
Can you point out an example of where arbitrary values are tested?

Reply:
> should be tested to make sure nothing weird happens!

But that is degenerate according to the API. By this argument we should also try `CustomQueryInterfaceMode.Allow + 2` and beyond.